turbopack/crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -496,6 +570,14 @@ impl FileSystem for DiskFileSystem {
         content: Vc<FileContent>,
     ) -> Result<Vc<Completion>> {
         let full_path = self.to_sys_path(fs_path).await?;
+
+        let full_path = validate_path_length(&full_path).with_context(|| {
+            format!(
+                "path length for file {} exceeds max length of filesystem",
+                full_path.to_string_lossy()
+            )
+        })?;
```

Comment:
Personally, I'd move this `.with_context` to line 146 inside of `validate_path_length` where you call `validate_path_length_inner`.